crates/telio-wg/src/wg.rs

```diff
@@ -44,6 +44,26 @@ use std::{
     time::Duration,
 };
 
+/// Interface for retrieving stats about network activity
+#[async_trait]
+pub trait ActivityRecorder: Sync + Send {
+    /// Retrieve global peer network activity timestamps
+    async fn get_latest_peer_network_activity(&self) -> Result<Option<TxRxTimestampPair>, Error>;
+}
```

Comment:
This trait could be omitted, however my rust-fu is limited here. session_keeper::start() is the place to look at for more context